specification.md

```diff
@@ -0,0 +1,191 @@
+     |             |
+ [Span D]      +---+-------+
+               |           |
+           [Span E]    [Span F] >>> [Span G] >>> [Span H]
+                                       ↑
+                                       ↑
+                                       ↑
+                         (Span G "FollowsFrom" Span F)
+
+~~~
+
+Sometimes it's easier to visualize **Traces** with a time axis as in the
+diagram below:
+
+~~~
+Temporal relationships between Spans in a single Trace
+
+
+––|–––––––|–––––––|–––––––|–––––––|–––––––|–––––––|–––––––|–> time
+
+ [Span A···················································]
+   [Span B··············································]
+      [Span D··········································]
+    [Span C········································]
+         [Span E·······]        [Span F··] [Span G··] [Span H··]
+~~~
+
+Each **Span** encapsulates the following state:
+
+- A start timestamp
+- A finish timestamp
+- A set of zero or more key:value **Span Tags**. The keys must be strings. The
+  values may be strings, bools, or numeric types.
+- A set of zero or more **Span Logs**, each of which is itself a key:value map
+  paired with a timestamp
+- A **SpanContext** (see below)
+- **References** to zero or more causally-related **Spans** (via the
+  **SpanContext** of those related **Spans**)
+
+Each **SpanContext** encapsulates the following state:
+
+- Any OpenTracing-implementation-dependent state (for example, trace and span ids) needed to refer to a distinct **Span** across a process boundary
+- **Baggage Items**, which are just key:value pairs that cross process boundaries
+
+## The OpenTracing API
+
+There are three critical and inter-related types in the OpenTracing specification: `Tracer`, `Span`, and `SpanContext`. Below, we go through the behaviors of each type; roughly speaking, each behavior becomes a "method" in a typical programming language, though it may actually be a set of related sibling methods due to type overloading and so on.
+
+When we discuss "optional" parameters, it is understood that different languages have different ways to construe such concepts. For example, in Go we might use the "functional Options" idiom, whereas in Java we might use a builder pattern.
+
+### `Tracer`
+
+The `Tracer` interface creates `Span`s and understands how to `Inject`
+(serialize) and `Extract` (deserialize) them across process boundaries.
+Formally, it has the following capabilities:
+
+#### Start a new `Span`
+
+Required parameters
+    - An **operation name**, a human-readable string which
+      concisely represents the work done by the Span (for example, an RPC
+      method name, a function name, or the name of a subtask or stage within a
+      larger computation). The operation name should be **the most general
+      string that identifies a (statistically) interesting class of `Span`
+      instances**. That is, `"get_user"` is better than `"get_user/314159"`.
```

Comment:
fwiw the current [spec](http://opentracing.io/documentation/pages/spec.html) had a table with 3 examples, maybe we can keep it.